### PR TITLE
Fixes #111

### LINKED ISSUE #111: Update question 153
### Thanks for helping!

Thank you for helping to port [this question](../blob/master/questions/153/) from C++17 to C++23.

You'll find the source code in [question.cpp](../blob/master/questions/153/question.cpp), this should normally not need modification.

In C++17, the correct answer is:
> The program has a compilation error

Please verify that this is still correct in C++23. If it is, please do the following:
- [ ] Update [explanation.md](../blob/master/questions/153/explanation.md):
  - [ ] Refer to the correct section numbers
  - [ ] Use updated quotes from those sections (the wording might have changed)
  - [ ] Make sure the rest of the text in the explanation is consistent with the new standard
- [ ] Update [hint.md](../blob/master/questions/153/hint.md) if needed (usually not needed)

*Note that some older questions are missing references to the standard, we don't need to fix that now.*

If you get stuck, please describe the problem in detail in a comment to the issue, and a maintainer will add the `help wanted` tag so others can chime in.

If the correct answer has changed from C++17 to C++23, you can either just leave a comment in this issue, or see the instructions for [updating meta data](../blob/master/METADATA_HOWTO.md).

### PR DESCRIPTION
**Pull request by knatten**
Fixes #111

### CODE REVIEW

File: questions/153/explanation.md [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
-
-The array of n const char decays to a pointer to const char. That pointer to const char does not implicitly convert to a pointer to non-const char.
-
-A note in §[conv.qual]¶4 extrapolates from the preceeding normative passages that "a prvalue of type “pointer to `cv1 T`” can be converted to a prvalue of type “pointer to `cv2 T`” if “`cv2 T`” is more cv-qualified than “`cv1 T`”." In this case however, `char*` is less cv-qualified than `const char *`, and the conversion is not allowed.
-
-Note: While most compilers still allow `char const[]` to `char*` conversion with just a warning, this is not a legal conversion since C++11.
-
+According to §[lex.string]¶8 in the standard:
+
+> A narrow string literal has type “array of *n* `const char`”
```
Comment (tocic) [inline questions/153/explanation.md:3]: I think we should reference §[lex.string]¶1 or §[tab:lex.string.literal] here

File: questions/153/explanation.md [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
+
+A note in §[conv.qual]¶4 extrapolates from the preceeding normative passages:
+
+> a prvalue of type “pointer to `cv1 T`” can be converted to a prvalue of type “pointer to `cv2 T`” if “`cv2 T`” is more cv-qualified than “`cv1 T`”.
+
+In this case however, `char*` is less cv-qualified than `const char *`, so the conversion is not allowed.
+
+Note: While most compilers still allow `char const[]` to `char*` conversion with just a warning, this is not a legal conversion since C++11.
+
 See also <http://dev.krzaq.cc/stop-assigning-string-literals-to-char-star-already/>
```
Comment (tocic) [inline questions/153/explanation.md:15]:
```suggestion
See also <https://dev.krzaq.cc/stop-assigning-string-literals-to-char-star-already/>
```

File: questions/153/explanation.md [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
-
+According to §[lex.string]¶8 in the standard:
+
+> A narrow string literal has type “array of *n* `const char`”
+
+The array of *n* `const char` decays to a pointer to `const char`. That pointer to `const char` does not implicitly convert to a pointer to non-const `char`.
+
+A note in §[conv.qual]¶4 extrapolates from the preceeding normative passages:
+
+> a prvalue of type “pointer to `cv1 T`” can be converted to a prvalue of type “pointer to `cv2 T`” if “`cv2 T`” is more cv-qualified than “`cv1 T`”.
```
Comment (tocic) [inline questions/153/explanation.md:9]:
```suggestion
§[conv.qual]¶note-3 extrapolates from the preceeding normative passages:

> A prvalue of type “pointer to *cv1* `T`” can be converted to a prvalue of type “pointer to *cv2* `T`” if “*cv2* `T`” is more cv-qualified than “*cv1* `T`”.
```

File: questions/153/explanation.md (hunk 1 above)
Comment (knatten) [reply questions/153/explanation.md]: Oh, good catch. Let's use the table.

**Review by tocic — APPROVED** [review]

**Pull request merged**